### ISSUE TITLE
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value'

### ISSUE THREAD

**Issue by sapphire444** [dyc3/steamguard-cli#253]
### Version

0.9.4 (macOS Ventura)

### What happened?

had to update from 0.7.0 to 0.9.4 in order to confirm my trade, appears something is wrong while migrating the manifest file to a newer version

### Relevant log output

```Shell
DEBUG - Args { global: GlobalArgs { username: None, all: false, mafiles_path: None, passkey: None, verbosity: Debug, no_update_check: false }, sub: None, code: CodeCommand { offline: false } }
INFO - reading manifest from /Users/user/Library/Application Support/steamguard-cli/maFiles
DEBUG - loading manifest: "/Users/user/Library/Application Support/steamguard-cli/maFiles/manifest.json"
INFO - Migrating manifest
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /Users/user/.cargo/registry/src/github.com-1ecc6299db9ec823/steamguard-cli-0.9.4/src/accountmanager/migrate.rs:25:16
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by dyc3** [comment]
It's failing on this

```rust
if path.extension().unwrap() == "maFile" {
    backup_file(&path).unwrap();
}
```

Would you happen to have any files listed in your manifest that don't have a file extension?

**Comment by sapphire444** [comment]
I don't have any files here than .maFile, manifest.json, and manifest.json.bak

**Comment by dyc3** [comment]
Could you open a terminal in that directory and run `ls -a -l`?

**Comment by sapphire444** [comment]
```
drwxr-xr-x  6 user *  192 Jun 30 01:04 .
drwxr-xr-x  4 user *   128 May 21 21:54 ..
-rw-r--r--@ 1 user  *  6148 Jun 30 01:04 .DS_Store
```

**Comment by sapphire444** [comment]
Seeing these and .maFile with jsons

**Comment by dyc3** [comment]
Ah, I bet it's the .DS_Store that's messing it up. According to the rust docs, extension will return: `None, if the file name begins with . and has no other .s within;` Fix will be in tonight's release.

**Comment by sapphire444** [comment]
Just tried to create maFiles and move maFiles over, seeing another error after trying to decrypt

```
`ERROR - Failed to migrate manifest: An unexpected error occurred during manifest migration: Failed to load some accounts: [DecryptionFailed(Unknown(Unpad Error))]
ERROR - An unexpected error occurred during manifest migration: Failed to load some accounts: [DecryptionFailed(Unknown(Unpad Error))]

Caused by:
    Failed to load some accounts: [DecryptionFailed(Unknown(Unpad Error))]`
```

Leaving it here because it could be related to exact the same issue supposedly
Or am I missing something?

**Comment by sapphire444** [comment]
Actually that was wrong pin, but seeing another error here

`ERROR - maFiles are encrypted, but no passkey was provided.
`

**Comment by dyc3** [comment]
That should also be fixed in the next release

**Comment by sapphire444** [comment]
Thank you so much :)

**Comment by dyc3** [comment]
Just published `v0.9.5`, open a new issue if you're still having issues.
